libs/core/src/lib/icon-tab-bar/components/icon-tab-bar.class.ts

```diff
@@ -0,0 +1,182 @@
+import {
+    ChangeDetectorRef,
+    Directive,
+    EventEmitter,
+    Input,
+    NgZone,
+    OnChanges, OnDestroy,
+    OnInit,
+    Output,
+    SimpleChanges,
+    ViewChild
+} from '@angular/core';
+import { IconTabBarItem, TabConfig } from '../types';
+import { cloneDeep } from '../../utils/functions/clone-deep';
+import { ICON_TAB_HIDDEN_CSS, UNIQUE_KEY_SEPARATOR } from '../constants';
+import { OverflowItemsDirective } from '../../utils/directives/overflow-items/overflow-items.directive';
+import { ExtraButtonDirective } from '../directives/extra-button/extra-button.directive';
+import { take, takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+@Directive()
+export abstract class IconTabBarClass implements OnInit, OnChanges, OnDestroy {
```

Comment:
Maybe it should be `IconTabBarBase`? Or any other name, but `..Class` not exposing the purpose.

Reply:
Add the TypeDoc comments please.